sdlib/d/gc/tcache.d

```diff
@@ -274,6 +284,40 @@ public:
 	}
 
 private:
+
+	CapacityInfo getCapacityInfo(PageDescriptor pd, void* ptr) {
+		auto e = pd.extent;
+
+		if (e.isLarge()) {
+			return CapacityInfo(e.address, e.size, e.usedCapacity);
+		}
+
+		// Slab alloc:
+		import d.gc.slab;
+		auto sg = SlabAllocGeometry(ptr, pd);
+		auto freeSize = e.getFreeSpace(sg.index);
```

Comment:
This will snoop in the extent to find out if freeSpace is supported, when it has all the required infos in the PageDescriptor. Clearly the API overall does not seems to be adapted to the problem at hand.

Reply:
Yes, but if the size class is not supported, then there is no point fetching data in the extent. the pageDescriptor has the size class, so it knows.

Reply:
This is why i was wondering in the first place if it made sense to allow this in the extent, as it would have to be a case where we come to the extent not from the pd, which I don't think happens ever at all.